bgpd: show bgp ipv4 label-unicast does not provide any output

dwalton76 opened the pull request:
Signed-off-by: Daniel Walton <dwalton@cumulusnetworks.com>
Reviewed-by:   Donald Sharp <sharpd@cumulusnetworks.com>

labeled-unicast routes are installed in the unicast table, to make
troubleshooting easier we need "show bgp ipv4 labeled-unicast x.x.x.x"
to show the x.x.x.x route in the unicast table.

eqvinox commented:
This could be very confusing for the user if "show ... labeled-unicast" prints output for a session which doesn't even have that SAFI enabled? (Mostly in case of the "show ... neighbor" command)

[not sure if we can do much to improve this... maybe just print a note like `labeled-unicast is treated as alias for unicast on this command` or something?]

dwalton76 commented:
@eqvinox just updated the PR with a few fixes
- `show bgp ipv4 labeled-unicast statistics` is working now
- `show bgp ipv4 labeled-unicast summary` was showing 0 in the `PfxRcd` column, fixed this
- `show bgp ipv4 labeled-unicast neighbor XYZ ...` was indeed broken, fixed this

```
cel-redxp-10# show bgp ipv4 unicast summ
BGP router identifier 6.0.0.7, local AS number 65200 vrf-id 0
BGP table version 57
RIB entries 101, using 14 KiB of memory
Peers 6, using 122 KiB of memory
Peer groups 2, using 128 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd
att(swp5)       4        100     111     114        0    0    0 00:04:52           20
twc(swp6)       4        200     119     122        0    0    0 00:04:57           20

Total number of neighbors 2
cel-redxp-10# 
cel-redxp-10# show bgp ipv4 labeled-unicast summ
BGP router identifier 6.0.0.7, local AS number 65200 vrf-id 0
BGP table version 0
RIB entries 0, using 0 bytes of memory
Peers 6, using 122 KiB of memory
Peer groups 2, using 128 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd
spine-1(swp1)   4      65000     113     111        0    0    0 00:05:04            6
spine-2(swp2)   4      65000     111     109        0    0    0 00:04:58            6
spine-3(swp3)   4      65000     113     111        0    0    0 00:05:04            6
spine-4(swp4)   4      65000     111     109        0    0    0 00:04:58            6

Total number of neighbors 4
cel-redxp-10# 
```

neighbor active for unicast but not labeled-unicast
```
cel-redxp-10# show bgp ipv4 unicast neighbor swp5 routes
BGP table version is 57, local router ID is 6.0.0.7
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 20.1.1.1/32      swp5            0             0 100 i
*> 20.1.1.2/32      swp5            0             0 100 i
*> 20.1.1.3/32      swp5            0             0 100 i
*> 20.1.1.4/32      swp5            0             0 100 i
*> 20.1.1.5/32      swp5            0             0 100 i
*> 20.1.1.6/32      swp5            0             0 100 i
*> 20.1.1.7/32      swp5            0             0 100 i
*> 20.1.1.8/32      swp5            0             0 100 i
*> 20.1.1.9/32      swp5            0             0 100 i
*> 20.1.1.10/32     swp5            0             0 100 i
*> 40.1.1.1/32      swp5            0             0 100 i
*> 40.1.1.2/32      swp5            0             0 100 i
*> 40.1.1.3/32      swp5            0             0 100 i
*> 40.1.1.4/32      swp5            0             0 100 i
*> 40.1.1.5/32      swp5            0             0 100 i
*> 40.1.1.6/32      swp5            0             0 100 i
*> 40.1.1.7/32      swp5            0             0 100 i
*> 40.1.1.8/32      swp5            0             0 100 i
*> 40.1.1.9/32      swp5            0             0 100 i
*> 40.1.1.10/32     swp5            0             0 100 i

Displayed  20 routes and 65 total paths
cel-redxp-10# 
cel-redxp-10# show bgp ipv4 labeled-unicast neighbor swp5 routes
% No such neighbor or address family
cel-redxp-10# 
```

neighbor active for labeled-unicast but not unicast
```
cel-redxp-10# show bgp ipv4 unicast neighbor swp1 routes
% No such neighbor or address family
cel-redxp-10# 
cel-redxp-10# show bgp ipv4 labeled-unicast neighbor swp1 routes
BGP table version is 57, local router ID is 6.0.0.7
Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
              i internal, r RIB-failure, S Stale, R Removed
Origin codes: i - IGP, e - EGP, ? - incomplete

   Network          Next Hop            Metric LocPrf Weight Path
*> 6.0.0.1/32       swp1            0             0 65000 i
*> 6.0.0.5/32       swp1                          0 65000 65100 i
*> 6.0.0.6/32       swp1                          0 65000 65100 i
*> 6.0.0.9/32       swp1                          0 65000 65100 65500 i
*> 6.0.0.10/32      swp1                          0 65000 65100 65500 i
*> 20.0.0.0         swp1            0             0 65000 i

Displayed  6 routes and 65 total paths
cel-redxp-10# 
```
